### PR TITLE
Add manitdplot settings to workbench - part 2

### LINKED ISSUE #27471: Workbench: Complete settings support
Pretty much everything from MPlot

Including (incomplete list):
- Default normalisation
- Hiding algorithm categories
- Hiding Fn categories
- Hiding Interfaces

### PR DESCRIPTION
**Pull request by alicerussell1**
~~**This PR should not be merged until #27787 has been merged**~~

**Description of work.**
This PR adds in mantidplot settings into workbench. This PR includes the 2 new tabs for fitting and plotting.

**To test:**
Open workbench
Click `file`->`settings`
Change the settings in the fitting and plotting tabs and check they work as expected. These tabs are new so all options need testing.

Fixes #27471 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by Harrietbrown — CHANGES_REQUESTED** [review]
It has occoured to me, it would be a good idea to have a note somewhere on the settings GUI letting users know that for some changes mantid will need to be restared to take effect. spesificaly in the `show/hide Catagories` tab. Sorry I should have thaught to this in the part 1.

**Comment by alicerussell1** [comment]
I have added a note at the bottom of the `show/hide categories` page and a tool tip on the few in `general` that require restarts.

**Comment by Harrietbrown** [comment]
Great, I'll Check that it all looks good and it will be ready to go.

**Review by Harrietbrown — DISMISSED** [review]
Good Stuff, all the options work and It's good to have the notifications that some settings need restarting to take effect.

File: qt/applications/workbench/workbench/widgets/settings/presenter.py [hunk 1]
```diff
@@ -12,26 +12,35 @@
...
+from workbench.widgets.settings.plots.presenter import PlotSettings
 from workbench.widgets.settings.view import SettingsView
 
 
 class SettingsPresenter(object):
     ASK_BEFORE_CLOSE_TITLE = "Confirm exit"
     ASK_BEFORE_CLOSE_MESSAGE = "Are you sure you want to exit without applying the settings?"
     SETTINGS_TABS = {'general_settings' : "General",
-                     'categories_settings' : "Show/Hide Categories"}
+                     'categories_settings' : "Show/Hide Categories",
```
Comment (martyngigg) [inline qt/applications/workbench/workbench/widgets/settings/presenter.py]: I know this is not strictly this PR but I think I missed the other one. It looks like "Show/Hide Categories" is always too wide for the width of the left-hand side box. Should we shorten this to just `Categories` to be consistent with the other single words? Perhaps we could add the Show /Hide to the "Algorithm Categories" & "Interface Categories" titles?

**Review by martyngigg — CHANGES_REQUESTED** [review]
This is good work. It's good that we can now do all this from workbench.

I don't know if this shows up in Windows but on starting workbench I get:

```$ bin/workbench                                                                            
/home/mgigg/Code/mantidproject/mantid-nightly/qt/applications/workbench/workbench/plotting/config.py:50: UserWarning:
This call to matplotlib.use() has no effect because the backend has already                      
been chosen; matplotlib.use() must be called *before* pylab, matplotlib.pyplot,
or matplotlib.backends is imported for the first time.                                                                  
                                                                    
The backend was *originally* set to 'TkAgg' by the following code:
  File "bin/workbench", line 11, in <module>               
    load_entry_point('MantidWorkbench', 'gui_scripts', 'workbench')()
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 480, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 2693, in load_entry_point
    return ep.load()
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 2324, in load
    return self.resolve()
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 2330, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/home/mgigg/Code/mantidproject/mantid-nightly/qt/applications/workbench/workbench/app/mainwindow.py", line 29, in <module>
    from workbench.widgets.settings.presenter import SettingsPresenter
  File "/home/mgigg/Code/mantidproject/mantid-nightly/qt/applications/workbench/workbench/widgets/settings/presenter.py", line 17, in <module>
    from workbench.widgets.settings.plots.presenter import PlotSettings
  File "/home/mgigg/Code/mantidproject/mantid-nightly/qt/applications/workbench/workbench/widgets/settings/plots/presenter.py", line 14, in <module>
    from workbench.plotting.figureinteraction import VALID_LINE_STYLE
  File "/home/mgigg/Code/mantidproject/mantid-nightly/qt/applications/workbench/workbench/plotting/figureinteraction.py", line 37, in <module>
    from workbench.plotting.toolbar import ToolbarStateManager
  File "/home/mgigg/Code/mantidproject/mantid-nightly/qt/applications/workbench/workbench/plotting/toolbar.py", line 13, in <module>
    from matplotlib.backends.backend_qt5 import NavigationToolbar2QT
  File "/usr/lib/python3/dist-packages/matplotlib/backends/__init__.py", line 14, in <module>
    line for line in traceback.format_stack()


  mpl.use(MPL_BACKEND)
FrameworkManager-[Notice] Welcome to Mantid 4.2.20200131.811
FrameworkManager-[Notice] Please cite: http://dx.doi.org/10.1016/j.nima.2014.07.029 and this release: http://dx.doi.org/10.5286/Software/Mantid
```

The reason is that matplotlib is quite sensitive to the order of importing wr.t. to certain other modules. I think what happens is the `plotting/figureinteraction.py` import causes the backend import to select the wrong backend at this point and later when the "correct" one needs setting it can't. It looks like we just want to constants from the `figureinteraction` module? Can I suggest we move the module constants out to separate module in `plotting`, something like `plotting.style`? It can just have the  `VALID_LINE_STYLE` and `VALID_COLORS` definitions for now and then the settings can import this instead?

One other thing I've also notice is that the settings has an "Ignore ParaView" setting - can we remove this? We are currently not porting ParaView so we don't want to make any suggestion that we are going to do that.

**Review by martyngigg — APPROVED** [review]
Thanks for making those changes. I get a clean startup now. :+1:

**Comment by martyngigg** [comment]
Test failure is unrelated.
